src/components/Autocomplete/Autocomplete.tsx

```diff
@@ -30,6 +30,8 @@ export interface Props {
   loading?: boolean;
   /** Indicates if more results will load dynamically */
   willLoadMoreResults?: boolean;
+  /** Is rendered when there are no options */
+  renderEmptyState?(): React.ReactElement<any>;
```

Comment:
I'm curious why it's a function?

Reply:
No real reason. It's pretty common in `web` to use a function here. But since we aren't passing it any params, it doesn't really matter.

The old autocomplete component also used a function here, so it would make it easier for consumers of the old autocomplete to migrate over to the new one.

Reply:
@AndrewMusgrave - You mean `renderEmptyState?(): React.ReactNode`? I'm cool with that.

Reply:
I agree, I think a React.ReactNode is sufficient and will accomplish what's needed without a function. Looking at empty states in lists on index pages in production, a consumer could pass an icon or a text container with a message.